sway-core/src/semantic_analysis/ast_node/expression/typed_expression.rs

```diff
@@ -367,8 +390,18 @@ impl TypedExpression {
                 }
                 buf
             }
+            TypedExpressionVariant::Reassignment(reassignment) => {
+                reassignment.rhs.gather_return_statements()
+            }
+            TypedExpressionVariant::StorageReassignment(_storage_reassignment) => {
+                // FIXME: we should check rhs here too, no?
+                vec![]
```

Comment:
This is in `fn gather_return_statements`. We don't check the rhs of storage assignments for return statements.

Reply:
Ditto on my above comment:
> yeah, technically the RHS could be a code block with a return in it, so we should check that.